frontend/packages/metal3-plugin/src/hooks/index.ts

```diff
@@ -1,9 +0,0 @@
-import { useEffect, useRef } from 'react';
-
-export const usePrevious = (value) => {
```

Comment:
@jtomasek I moved this hook to `shared` and also simplified it a bit.

Reply:
FIY useEffect is async and storing value to ref wont trigger a component update. Which means for first render this hook returns undefined and second render (which have to be triggered by something else) will return the correct value.

Reply:
I think its usually fine. But there's no way to update the previous value (after all its called previous, not first). Once you try add dependency array to be able to update the initial value the async nature of useEffect will cause issues.